.github/workflows/benchmarks.yml

```diff
@@ -0,0 +1,56 @@
+name: Benchmarks
+
+on:
+  push:
+    branches: ["main"]
+
+  pull_request:
+    branches: ["main"]
+
+jobs:
+  benchmark:
+    name: Benchmarks
+    strategy:
+      matrix:
+        go-version: ["1.18"]
+        platform: ["ubuntu-latest"]
+
+    runs-on: ${{ matrix.platform }}
+    timeout-minutes: 10
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b
+
+      - name: Set up Go ${{ matrix.go-version }}
+        uses: actions/setup-go@268d8c0ca0432bb2cf416faae41297df9d262d7f
+        with:
+          go-version: ${{ matrix.go-version }}
+        id: go
+
+      - name: Run benchmark
+        run: go test  -benchmem -run="^$" -bench "^Benchmark" . github.com/timbray/quamina | tee output.txt
+
+      - name: Download previous benchmark data
+        uses: actions/cache@v1
```

Comment:
What is the default of how long the cache content is retained? E.g. if we don't have PRs and updated benchmark data pushed to the cache, do we lose the data at some point and then miss regressions for some reasons?

Reply:
According to github, the cache will be maintained for 7 days if it wasn't accessed. I am not sure what will be the behaviour if we don't touch quamina's main branch for more than 7 days.

https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#usage-limits-and-eviction-policy